cpu/cortexm_common/vectors_cortexm.c

```diff
@@ -110,6 +111,52 @@ void nmi_default(void)
 
 void hard_fault_default(void)
```

Comment:
>In general yes, but why not staying inside the naked function?

Naked functions are only allowed to contain inline asm in some compilers (e.g. Clang-3.6), see https://github.com/RIOT-OS/RIOT/pull/2669